web/profiles/joinup/joinup.behat.inc

```diff
@@ -105,28 +105,27 @@ class JoinupSubContext extends DrupalSubContextBase implements DrupalSubContextI
    */
   public function submitRegistrationForm(TableNode $table, $location) {
     $this->visitPath($location);
+    $username = '';
 
     // Fill in fields.
     foreach ($table->getRowsHash() as $field => $value) {
       $this->getSession()->getPage()->fillField($field, $value);
+      if ($field == 'Username') {
+        $username = $value;
+      }
     };
     // Submit form.
     $this->getSession()->getPage()->pressButton("Create new account");
 
     // Get the last registered user.
-    $storage = \Drupal::entityManager()->getStorage('user');
-    $query = $storage->getQuery();
-    $result = $query->sort('uid', 'DESC')->range(0, 1)->execute();
-    $uid = array_pop($result);
+    $results = \Drupal::entityManager()->getStorage('user')
+        ->loadByProperties(['name' => $username]);
```

Comment:
Well, I don't think we need to worry about this. This code runs after a registration page is submitted. If the username is not passed at this time, this would be a bug of core I guess (or some other implementation that allowed it). Should we still catch an exception for that possibility?